api/src/test/java/com/epam/pipeline/manager/pipeline/RunScheduleManagerTest.java

```diff
@@ -103,214 +115,240 @@ public void setUp() {
+                .when(mockRunScheduleDao).loadAllRunSchedulesBySchedulableIdAndType(RUN_ID, ScheduleType.PIPELINE_RUN);
 
         runScheduleManager.deleteSchedules(RUN_ID, ScheduleType.PIPELINE_RUN);
-        final List<RunSchedule> loadRunSchedule = runScheduleManager
-                .loadAllSchedulesBySchedulableId(RUN_ID, ScheduleType.PIPELINE_RUN);
-        assertEquals(0, loadRunSchedule.size());
-        Mockito.verify(runScheduler, Mockito.times(2)).unscheduleRunSchedule(any());
 
+        verify(mockRunScheduleDao, times(1)).deleteRunSchedules(anyLong(), any());
+        verify(mockRunScheduler, times(2)).unscheduleRunSchedule(any());
     }
 
     @Test
-    @WithMockUser(username = USER_OWNER)
     public void testDeleteRunSchedulesForPipeline() {
-        runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
-                Arrays.asList(testRunScheduleVO, testRunScheduleVO2));
-        Mockito.verify(runScheduler, Mockito.times(2)).scheduleRunSchedule(any());
+        doReturn(
+                Arrays.asList(
+                        createRunSchedule(1L, RUN_ID, RunScheduledAction.PAUSE, CRON_EXPRESSION1),
+                        createRunSchedule(2L, RUN_ID, RunScheduledAction.RESUME, CRON_EXPRESSION2)))
+                .when(mockRunScheduleDao).loadAllRunSchedulesBySchedulableIdAndType(RUN_ID, ScheduleType.PIPELINE_RUN);
 
-        runScheduleManager.deleteSchedulesForRunByPipeline(testPipeline.getId());
+        doReturn(Arrays.asList(createPipelineRun(RUN_ID, testPipeline.getId())))
+                .when(mockPipelineRunManager).loadAllRunsByPipeline(testPipeline.getId());
 
-        final List<RunSchedule> loadRunSchedule = runScheduleManager
-                .loadAllSchedulesBySchedulableId(RUN_ID, ScheduleType.PIPELINE_RUN);
-        assertEquals(0, loadRunSchedule.size());
-        Mockito.verify(runScheduler, Mockito.times(2)).unscheduleRunSchedule(any());
+        runScheduleManager.deleteSchedulesForRunByPipeline(testPipeline.getId());
 
+        verify(mockPipelineRunManager, times(1)).loadAllRunsByPipeline(anyLong());
+        verify(mockRunScheduleDao, times(1)).loadAllRunSchedulesBySchedulableIdAndType(anyLong(), eq(ScheduleType.PIPELINE_RUN));
+        verify(mockRunScheduler, times(2)).unscheduleRunSchedule(any());
+        verify(mockRunScheduleDao, times(1)).deleteRunSchedulesForRunByPipeline(anyLong());
     }
 
     @Test(expected = IllegalArgumentException.class)
-    @WithMockUser(username = USER_OWNER)
     public void loadRunScheduleByWrongId() {
-        final RunSchedule runSchedule =
-            runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
-                    Collections.singletonList(testRunScheduleVO)).get(0);
-        runScheduleManager.loadSchedule(runSchedule.getSchedulableId());
+        doThrow(IllegalArgumentException.class).when(mockRunScheduleDao).loadRunSchedule(1L);
+
+        runScheduleManager.loadSchedule(1L);
```

Comment:
You can see that according to `mockRunScheduleDao::loadRunSchedule` implementation it doesn't throw errors on missing run schedules but rather return an empty optional.